From 60213a107b4f7aee720010df8aad776509f6e43d Mon Sep 17 00:00:00 2001
From: Antonio Alvarez Feijoo <antonio.feijoo@suse.com>
Date: Fri, 3 Jul 2026 16:15:10 +0200
Subject: [PATCH 17/20] perf(devicetree-firmware): do not call inst_multiple if
 there are no fw files

Upstream status: cherry-pick from upstream commit c1cb94f082b8.

Otherwise `inst_multiple -o` is called without arguments.
---
 modules.d/70devicetree-firmware/module-setup.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/70devicetree-firmware/module-setup.sh b/modules.d/70devicetree-firmware/module-setup.sh
index 5b6b7467..100e027a 100755
--- a/modules.d/70devicetree-firmware/module-setup.sh
+++ b/modules.d/70devicetree-firmware/module-setup.sh
@@ -40,7 +40,7 @@ install_hostonly() {
             done
         done
     done
-    inst_multiple -o "${_fw_files[@]}"
+    ((${#_fw_files[@]} > 0)) && inst_multiple -o "${_fw_files[@]}"
 }
 
 # generic install (hostonly / generic use completely different approaches)
@@ -58,7 +58,7 @@ install_generic() {
             done
         done
     done
-    inst_multiple -o "${_fw_files[@]}"
+    ((${#_fw_files[@]} > 0)) && inst_multiple -o "${_fw_files[@]}"
 }
 
 # called by dracut
-- 
2.55.0

